app/src/main/java/com/app/foodforyou/view/MainActivity.java

```diff
@@ -154,7 +154,7 @@ private void getMainCategoryResponse() {
     }
 
     private void onClickCategory() {
-        intent = new Intent(getApplicationContext(), RecommendDietListActivity.class);
+        intent = new Intent(getApplicationContext(), DietListActivity.class);
```

Comment:
application context is tied to Application and its lifecycle. So should be used in such cases:

- Load Resource Values
- Start a Service
- Bind to a Service
- Send a Broadcast
- Register BroadcastReceiver

So for intent, use activity instead.

Reply:
Thank you for your explaining in detail. :) 
I've learned about service but I still confused. so could you make sure the way I thought is correct? 

- Load Resource Values -> this should be run using **getMainCategoryResponse** method in **MainActivity**
- Start a Service -> Make new Service class (ex. GetMainCategoryService)
- Bind to a Service -> In new service class, it should be bind with DietListActivity.

If I miss or wrong something, please let me know. Thank you!

Reply:
Ahh small misunderstanding Stella. I was just explaining the cases when applicationContext should be used :)

Basically if you dont need to use service or broadcast receiver in app then no need to use applicationContext. Just simple context or activity is enough.

This table is good summary of when to use.

![context](https://user-images.githubusercontent.com/23381115/94355309-25080180-00be-11eb-82d8-96b5920b5cbe.png)

Reply:
got it thank you! I think part of network is related with service. I'll make new issue and commit.
Thank you! 👍